Improve behaviour, extend functionality, refactor using ES6, add new development/build system

lucafalasco opened the pull request:
#### Additional features:
* Allows multiple listeners to be registerd on different DOM elements
* Improves show/hide behaviour
* Improves development and build system
* ES6 syntax

lucafalasco committed:
upgrade build system, add new eslint rules: ES6 | babel (stage-0) | webpack | yarn

lucafalasco committed:
refactor code with ES6 syntax and new linter rules

lucafalasco committed:
add new folder structure, rename files

lucafalasco committed:
update README

lucafalasco committed:
add minified package

lucafalasco commented:
Signed 🙂

kennethormandy commented:
Hey thanks so much! The fix in 63be31b and c2bd14c are both nice improvements that I’d be happy to include.

A couple of other comments:

- Initially I was a little hesitant to set up the whole Webpack/ES6 environment since I didn’t really think I’d be using it, but I can see why it makes sense going forward. That said, it increases the bundle size from **18kb** (7.2kb gzipped) which is primarily the SVG coordinates, to **87kb** (31.1kb gzipped). This isn’t really a problem for other projects using Webpack, but ideally this badge can be dropped into a project that’s even just static HTML and CSS, so it might be better to keep the bundle size small. What do you think?
- Can we keep the `npm install`/`npm run` parts in the README? I don’t think it’s a problem to have the `yarn` stuff as well, but we might as well keep the existing instructions since they still work.
- ~~The `webpack-dev-server` seems to work if you run it manually, but not as part of the `start` run script right now.~~ I think this is fine actually.
- Can we use an approach to cleaning the build folder other than `rm`?

Since originally writing this, Google updated the ESLint package to use a different approach for ES6 things, but we can always open another PR to deal with that.

I quickly ran this on [googlefonts/japanese](https://github.com/googlefonts/japanese) and it seems fine to me. I also published a version of the test page if anyone else needs to look it over:

- Before http://friends-badge-before-pr9.surge.sh/test
- After http://friends-badge-pr9.surge.sh/

Thanks again for this @lucafalasco

davelab6 commented:
Thanks @lucafalasco :D

lucafalasco committed:
reduce bundle size

lucafalasco committed:
revert to previous create/remove folder method in build script

lucafalasco committed:
change bundle folder

lucafalasco committed:
some refactor & cleanup

lucafalasco committed:
update README

lucafalasco commented:
Hi @kennethormandy and @davelab6, glad to know you appreciated ;)

I completely agree with your comments and updated the PR: 

* The bundle size problem should be fixed (apparently it was caused by importing the throttle function from the main lodash module..)

* Also, I reintroduced the npm commands in the README, and added ```yarn``` only at the beginning as an option, I think you're right and it makes more sense like this, also the other commands are basically the same so no need to keep both..

* Creating/cleaning dist folder works just fine as you did it, actually no reason to change it so I reverted it.
